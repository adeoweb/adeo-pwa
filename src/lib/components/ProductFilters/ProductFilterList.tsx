import React, { FunctionComponent } from 'react';

import filterOutNullableValues from 'src/peregrine/lib/util/adeoweb/filterOutNullableValues';

import { TProductFilterProps } from './ProductFilterTypes';

const ProductFilterList: FunctionComponent<TProductFilterProps> = ({
    filter,
    isActive,
    toggleOption
}) => {
    const options = filterOutNullableValues(filter?.options);

    if (!options.length) {
        return null;
    }

    const getIcon = (value: string) =>
        isActive(value) ? (
            <i className="icon-check" />
        ) : (
            <i className="icon-check-empty" />
        );

    return (
        <ul className="cat-list">
            {options.map(option => {
                const { value, count, label } = option;
                const icon = getIcon(value);
                const text = `${label}${count ? ' (' + count + ')' : ''}`;

                return (
                    <li key={value}>
                        <button onClick={() => toggleOption(value)}>
                            {icon}
                            <span dangerouslySetInnerHTML={{ __html: text }} />
                        </button>
                    </li>
                );
            })}
        </ul>
    );
};

ProductFilterList.displayName = 'ProductFilterList';

export default ProductFilterList;
