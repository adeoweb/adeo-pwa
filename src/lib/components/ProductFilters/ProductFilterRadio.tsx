import React, { FunctionComponent, useCallback } from 'react';
import { useTranslation } from 'react-i18next';
import { TProductFilterProps } from './ProductFilterTypes';
import { TAggregationOption } from 'src/lib/types/graphql/Aggregation';
import filterOutNullableValues from 'src/peregrine/lib/util/adeoweb/filterOutNullableValues';

const ProductFilterRadio: FunctionComponent<TProductFilterProps> = ({
    filter,
    isActive,
    setOptions
}) => {
    const options = filterOutNullableValues(filter?.options);
    const { t } = useTranslation();
    const selectValue = useCallback(
        (value: string) => {
            setOptions([value]);
        },
        [setOptions]
    );

    if (!options.length) {
        return null;
    }

    const getIcon = (value: string) =>
        isActive(value) ? (
            <i className="icon-check" />
        ) : (
            <i className="icon-check-empty" />
        );

    const getLabel = (option: TAggregationOption) => {
        const { value, label } = option;
        switch (value) {
            case '0':
                return t('No');
            case '1':
                return t('Yes');
            default:
                return label;
        }
    };

    return (
        <ul className="cat-list">
            {options.map(option => {
                const { value, count } = option;
                const label = getLabel(option);
                const icon = getIcon(value);
                const text = `${label}${count ? ' (' + count + ')' : ''}`;

                return (
                    <li key={value}>
                        <button onClick={() => selectValue(value)}>
                            {icon}
                            {text}
                        </button>
                    </li>
                );
            })}
        </ul>
    );
};

ProductFilterRadio.displayName = 'ProductFilterRadio';

export default ProductFilterRadio;
