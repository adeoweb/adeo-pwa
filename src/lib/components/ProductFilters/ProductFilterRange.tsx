import React, { FunctionComponent } from 'react';
import { TProductFilterProps } from './ProductFilterTypes';

const ProductFilterRange: FunctionComponent<TProductFilterProps> = () => {
    // TODO: range widget filter
    return (
        <form action="/">
            <div className="price-slider-wrapper">
                <div
                    id="price-slider"
                    className="noUi-target noUi-ltr noUi-horizontal"
                >
                    <div className="noUi-base">
                        <div className="noUi-connects">
                            <div
                                className="noUi-connect"
                                style={{
                                    transform:
                                        'translate(20%, 0px) scale(0.5, 1)'
                                }}
                            />
                        </div>
                        <div
                            className="noUi-origin"
                            style={{
                                transform: 'translate(-80%, 0px)',
                                zIndex: 5
                            }}
                        >
                            <div
                                className="noUi-handle noUi-handle-lower"
                                data-handle="0"
                                tabIndex={0}
                                role="slider"
                                aria-orientation="horizontal"
                                aria-valuemin={0.0}
                                aria-valuemax={60.0}
                                aria-valuenow={20.0}
                                aria-valuetext="200.00"
                            />
                        </div>
                        <div
                            className="noUi-origin"
                            style={{
                                transform: 'translate(-30%, 0px)',
                                zIndex: 4
                            }}
                        >
                            <div
                                className="noUi-handle noUi-handle-upper"
                                data-handle="1"
                                tabIndex={0}
                                role="slider"
                                aria-orientation="horizontal"
                                aria-valuemin={30.0}
                                aria-valuemax={100.0}
                                aria-valuenow={70.0}
                                aria-valuetext="700.00"
                            />
                        </div>
                    </div>
                </div>
            </div>

            <div className="filter-price-action">
                <button type="submit" className="btn btn-primary">
                    Filter
                </button>

                <div className="filter-price-text">
                    Price:
                    <span id="filter-price-range">$200.00 - $700.00</span>
                </div>
            </div>
        </form>
    );
};

ProductFilterRange.displayName = 'ProductFilterRange';

export default ProductFilterRange;
