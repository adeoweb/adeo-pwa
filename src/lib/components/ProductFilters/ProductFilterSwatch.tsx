import React, { FunctionComponent } from 'react';

import filterOutNullableValues from 'src/peregrine/lib/util/adeoweb/filterOutNullableValues';

import { TProductFilterProps } from './ProductFilterTypes';

const ProductFilterSwatch: FunctionComponent<TProductFilterProps> = ({
    filter,
    isActive,
    toggleOption
}) => {
    const options = filterOutNullableValues(filter.options);

    if (!options || !options.length) {
        return null;
    }

    return (
        <ul className="config-swatch-list">
            {options.map(option => {
                const { value, count, label } = option;
                const style = {
                    // TODO: temp, representational solution
                    // refactor to use actual color code/image from backend
                    backgroundColor: (label || '').toLowerCase()
                };
                const text = `${label}${count ? ' (' + count + ')' : ''}`;

                return (
                    <li key={value} className={isActive(value) ? 'active' : ''}>
                        <button onClick={() => toggleOption(value)}>
                            <span className="color-panel" style={style} />
                            <span>{text}</span>
                        </button>
                    </li>
                );
            })}
        </ul>
    );
};

ProductFilterSwatch.displayName = 'ProductFilterSwatch';

export default ProductFilterSwatch;
