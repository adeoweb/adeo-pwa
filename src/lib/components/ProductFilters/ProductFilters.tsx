import React, { Fragment, FunctionComponent } from 'react';

import { TAggregation } from 'src/lib/types/graphql/Aggregation';

import ProductFilterWrapper from './ProductFilterWrapper';

export type TProductFiltersProps = {
    filters: TAggregation[];
    activeFilters: Map<string, string[]>;
    setFilter: (attributeCode: string, values: string[]) => void;
    toggleFilter: (attributeCode: string, value: string) => void;
};

const ProductFilters: FunctionComponent<TProductFiltersProps> = ({
    filters,
    activeFilters,
    setFilter,
    toggleFilter
}) => {
    if (!filters.length) {
        return null;
    }

    return (
        <Fragment>
            {filters.map(filter => {
                const { attribute_code: attributeCode } = filter;
                const filterProps = {
                    filter: filter,
                    activeOptions: activeFilters.get(attributeCode) || [],
                    setFilter,
                    toggleFilter
                };
                return (
                    <ProductFilterWrapper
                        key={attributeCode}
                        {...filterProps}
                    />
                );
            })}
        </Fragment>
    );
};

export default ProductFilters;
