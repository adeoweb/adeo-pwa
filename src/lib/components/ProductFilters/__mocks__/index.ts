export {
    activeFiltersMock,
    setFilterMock,
    toggleFilterMock,
    filtersMock,
    productFilterPropsMock
} from './ProductFilters.mock';
