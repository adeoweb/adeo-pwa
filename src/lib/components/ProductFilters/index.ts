export { default } from './ProductFilters';
export { ProductFilterType } from './ProductFilterTypes';
