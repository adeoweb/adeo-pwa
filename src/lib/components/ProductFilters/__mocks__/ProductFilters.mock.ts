import { TProductFiltersProps } from '../ProductFilters';

export const activeFiltersMock = new Map([['category_id', ['3']]]);
export const setFilterMock = () => {};
export const toggleFilterMock = () => {};

export const filtersMock = [
    {
        attribute_code: 'price',
        count: 7,
        label: 'Price',
        options: [
            {
                count: 12,
                label: '20-30',
                value: '20_30'
            },
            {
                count: 15,
                label: '30-40',
                value: '30_40'
            },
            {
                count: 4,
                label: '40-50',
                value: '40_50'
            },
            {
                count: 10,
                label: '50-60',
                value: '50_60'
            },
            {
                count: 5,
                label: '60-70',
                value: '60_70'
            },
            {
                count: 3,
                label: '70-80',
                value: '70_80'
            },
            {
                count: 1,
                label: '80-*',
                value: '80_*'
            }
        ]
    },
    {
        attribute_code: 'category_id',
        count: 11,
        label: 'Category',
        options: [
            {
                count: 13,
                label: 'New Luma Yoga Collection',
                value: '8'
            },
            {
                count: 50,
                label: 'Tops',
                value: '21'
            },
            {
                count: 12,
                label: 'Jackets',
                value: '23'
            },
            {
                count: 12,
                label: 'Hoodies & Sweatshirts',
                value: '24'
            },
            {
                count: 12,
                label: 'Tees',
                value: '25'
            },
            {
                count: 14,
                label: 'Bras & Tanks',
                value: '26'
            },
            {
                count: 12,
                label: 'Women Sale',
                value: '30'
            },
            {
                count: 12,
                label: 'Tees',
                value: '33'
            },
            {
                count: 11,
                label: 'Erin Recommends',
                value: '34'
            },
            {
                count: 13,
                label: 'Performance Fabrics',
                value: '35'
            },
            {
                count: 6,
                label: 'Eco Friendly',
                value: '36'
            }
        ]
    },
    {
        attribute_code: 'color',
        count: 10,
        label: 'Color',
        options: [
            {
                count: 12,
                label: 'Black',
                value: '5476'
            },
            {
                count: 22,
                label: 'Blue',
                value: '5477'
            },
            {
                count: 1,
                label: 'Brown',
                value: '5478'
            },
            {
                count: 8,
                label: 'Gray',
                value: '5479'
            },
            {
                count: 18,
                label: 'Green',
                value: '5480'
            },
            {
                count: 21,
                label: 'Orange',
                value: '5483'
            },
            {
                count: 23,
                label: 'Purple',
                value: '5484'
            },
            {
                count: 18,
                label: 'Red',
                value: '5485'
            },
            {
                count: 9,
                label: 'White',
                value: '5486'
            },
            {
                count: 16,
                label: 'Yellow',
                value: '5487'
            }
        ]
    },
    {
        attribute_code: 'eco_collection',
        count: 2,
        label: 'Eco Collection',
        options: [
            {
                count: 38,
                label: '0',
                value: '0'
            },
            {
                count: 12,
                label: '1',
                value: '1'
            }
        ]
    }
];
export const productFilterPropsMock: TProductFiltersProps = {
    activeFilters: activeFiltersMock,
    setFilter: setFilterMock,
    toggleFilter: toggleFilterMock,
    filters: filtersMock
};
