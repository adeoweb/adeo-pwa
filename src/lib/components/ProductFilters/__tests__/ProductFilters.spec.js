import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import testRenderer from 'react-test-renderer';
import { productFilterPropsMock } from 'src/lib/components/ProductFilters/__mocks__';
import ProductFilters from '../ProductFilters';

test('Renders correctly', () => {
    const component = testRenderer.create(
        <Router>
            <ProductFilters {...productFilterPropsMock} />
        </Router>
    );
    expect(component.toJSON()).toMatchSnapshot();
});
