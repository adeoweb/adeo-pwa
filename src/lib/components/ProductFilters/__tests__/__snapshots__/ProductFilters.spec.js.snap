// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
Array [
  <div
    className="widget"
  >
    <h3
      className="widget-title"
    >
      <button
        aria-expanded={true}
        className=""
        onClick={[Function]}
      >
        Category
      </button>
    </h3>
    <div
      className="collapse show"
    >
      <div
        className="widget-body"
      >
        <ul
          className="cat-list"
        >
          <li>
            <button
              onClick={[Function]}
            >
              <i
                className="icon-check-empty"
              />
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "New Luma Yoga Collection (13)",
                  }
                }
              />
            </button>
          </li>
          <li>
            <button
              onClick={[Function]}
            >
              <i
                className="icon-check-empty"
              />
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "Tops (50)",
                  }
                }
              />
            </button>
          </li>
          <li>
            <button
              onClick={[Function]}
            >
              <i
                className="icon-check-empty"
              />
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "Jackets (12)",
                  }
                }
              />
            </button>
          </li>
          <li>
            <button
              onClick={[Function]}
            >
              <i
                className="icon-check-empty"
              />
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "Hoodies & Sweatshirts (12)",
                  }
                }
              />
            </button>
          </li>
          <li>
            <button
              onClick={[Function]}
            >
              <i
                className="icon-check-empty"
              />
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "Tees (12)",
                  }
                }
              />
            </button>
          </li>
          <li>
            <button
              onClick={[Function]}
            >
              <i
                className="icon-check-empty"
              />
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "Bras & Tanks (14)",
                  }
                }
              />
            </button>
          </li>
          <li>
            <button
              onClick={[Function]}
            >
              <i
                className="icon-check-empty"
              />
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "Women Sale (12)",
                  }
                }
              />
            </button>
          </li>
          <li>
            <button
              onClick={[Function]}
            >
              <i
                className="icon-check-empty"
              />
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "Tees (12)",
                  }
                }
              />
            </button>
          </li>
          <li>
            <button
              onClick={[Function]}
            >
              <i
                className="icon-check-empty"
              />
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "Erin Recommends (11)",
                  }
                }
              />
            </button>
          </li>
          <li>
            <button
              onClick={[Function]}
            >
              <i
                className="icon-check-empty"
              />
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "Performance Fabrics (13)",
                  }
                }
              />
            </button>
          </li>
          <li>
            <button
              onClick={[Function]}
            >
              <i
                className="icon-check-empty"
              />
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "Eco Friendly (6)",
                  }
                }
              />
            </button>
          </li>
        </ul>
      </div>
    </div>
  </div>,
  <div
    className="widget"
  >
    <h3
      className="widget-title"
    >
      <button
        aria-expanded={true}
        className=""
        onClick={[Function]}
      >
        Color
      </button>
    </h3>
    <div
      className="collapse show"
    >
      <div
        className="widget-body"
      >
        <ul
          className="config-swatch-list"
        >
          <li
            className=""
          >
            <button
              onClick={[Function]}
            >
              <span
                className="color-panel"
                style={
                  Object {
                    "backgroundColor": "black",
                  }
                }
              />
              <span>
                Black (12)
              </span>
            </button>
          </li>
          <li
            className=""
          >
            <button
              onClick={[Function]}
            >
              <span
                className="color-panel"
                style={
                  Object {
                    "backgroundColor": "blue",
                  }
                }
              />
              <span>
                Blue (22)
              </span>
            </button>
          </li>
          <li
            className=""
          >
            <button
              onClick={[Function]}
            >
              <span
                className="color-panel"
                style={
                  Object {
                    "backgroundColor": "brown",
                  }
                }
              />
              <span>
                Brown (1)
              </span>
            </button>
          </li>
          <li
            className=""
          >
            <button
              onClick={[Function]}
            >
              <span
                className="color-panel"
                style={
                  Object {
                    "backgroundColor": "gray",
                  }
                }
              />
              <span>
                Gray (8)
              </span>
            </button>
          </li>
          <li
            className=""
          >
            <button
              onClick={[Function]}
            >
              <span
                className="color-panel"
                style={
                  Object {
                    "backgroundColor": "green",
                  }
                }
              />
              <span>
                Green (18)
              </span>
            </button>
          </li>
          <li
            className=""
          >
            <button
              onClick={[Function]}
            >
              <span
                className="color-panel"
                style={
                  Object {
                    "backgroundColor": "orange",
                  }
                }
              />
              <span>
                Orange (21)
              </span>
            </button>
          </li>
          <li
            className=""
          >
            <button
              onClick={[Function]}
            >
              <span
                className="color-panel"
                style={
                  Object {
                    "backgroundColor": "purple",
                  }
                }
              />
              <span>
                Purple (23)
              </span>
            </button>
          </li>
          <li
            className=""
          >
            <button
              onClick={[Function]}
            >
              <span
                className="color-panel"
                style={
                  Object {
                    "backgroundColor": "red",
                  }
                }
              />
              <span>
                Red (18)
              </span>
            </button>
          </li>
          <li
            className=""
          >
            <button
              onClick={[Function]}
            >
              <span
                className="color-panel"
                style={
                  Object {
                    "backgroundColor": "white",
                  }
                }
              />
              <span>
                White (9)
              </span>
            </button>
          </li>
          <li
            className=""
          >
            <button
              onClick={[Function]}
            >
              <span
                className="color-panel"
                style={
                  Object {
                    "backgroundColor": "yellow",
                  }
                }
              />
              <span>
                Yellow (16)
              </span>
            </button>
          </li>
        </ul>
      </div>
    </div>
  </div>,
  <div
    className="widget"
  >
    <h3
      className="widget-title"
    >
      <button
        aria-expanded={true}
        className=""
        onClick={[Function]}
      >
        Eco Collection
      </button>
    </h3>
    <div
      className="collapse show"
    >
      <div
        className="widget-body"
      >
        <ul
          className="cat-list"
        >
          <li>
            <button
              onClick={[Function]}
            >
              <i
                className="icon-check-empty"
              />
              No (38)
            </button>
          </li>
          <li>
            <button
              onClick={[Function]}
            >
              <i
                className="icon-check-empty"
              />
              Yes (12)
            </button>
          </li>
        </ul>
      </div>
    </div>
  </div>,
]
`;
